### PR TITLE
 DMP-2183 throw exception when register invoked with null service con…

### PR DESCRIPTION
**Pull request by mario-paniccia**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-2183

### Change description ###
In the case of authentication with username and pwd from body, the existing behaviour of throwing `NoIdentitiesFoundException` is kept if service context is missing from the register request body. 
For all other types of authentication
we've introduced a new `NullServiceContextException` if service context is missing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/ws/ContextRegistryEndpoint.java [hunk 1]
```diff
@@ -68,4 +73,10 @@ public JAXBElement<LookupResponse> lookup(@RequestPayload JAXBElement<documentum
 
         return new ObjectFactory().createLookupResponse(lookupResponse);
     }
+
+    private static void validateRegisterBodyServiceContextIsNotNull(JAXBElement<Register> register) {
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/ws/ContextRegistryEndpoint.java:77]: Maybe we could push this into a validator?

**Review by mestebanez — APPROVED** [review]
@mario-paniccia I have approved. Left 2 comments if you are interested in addressing

File: src/main/java/uk/gov/hmcts/darts/ws/ContextRegistryEndpoint.java (hunk 1 above)
Comment (mario-paniccia) [reply src/main/java/uk/gov/hmcts/darts/ws/ContextRegistryEndpoint.java:77]: not sure how much worth is introducing a dedicated validator Michael for only this simple check. I would agree with you if we had plenty of different validations needed. Let's refactor to a dedicated validator if we introduced additional validations in the future

File: src/integrationTest/java/uk/gov/hmcts/darts/ws/ContextRegistryJwtServiceTest.java [hunk 2]
```diff
@@ -157,6 +157,14 @@ void testHandleRegisterWithAuthenticationToken(ContextRegistryClient client) thr
         }, getContextClient(), getGatewayUri(), DEFAULT_HEADER_USERNAME, DEFAULT_HEADER_PASSWORD);
     }
 
+    @ParameterizedTest
```
[the remark this answers is not in the document]
Comment (mario-paniccia) [reply src/integrationTest/java/uk/gov/hmcts/darts/ws/ContextRegistryJwtServiceTest.java:160]: ok, for username/password the change applies to the hear username/pwd validation, I'll add a test for it too 👍

[the remark this answers is not in the document]
Comment (mario-paniccia) [reply src/integrationTest/java/uk/gov/hmcts/darts/ws/ContextRegistryJwtServiceTest.java:160]: agreed with Michael that this is no longer needed

**Pull request merged**